polkadot/runtime/parachains/src/runtime_api_impl/vstaging.rs

```diff
@@ -15,3 +15,93 @@
 // along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
 
 //! Put implementations of functions from staging APIs here.
+
+use crate::{
+	configuration, dmp, hrmp, inclusion, initializer, paras,
+	shared,
+};
+use frame_support::traits::{GetStorageVersion, StorageVersion};
+use frame_system::pallet_prelude::*;
+use polkadot_primitives::{
+	async_backing::{
+		InboundHrmpLimitations, OutboundHrmpChannelLimitations,
+	},
+	vstaging::{
+		async_backing::Constraints,
+	},
+	Id as ParaId,
+};
+
+/// Implementation for `constraints` function from the runtime API
+pub fn constraints<T: initializer::Config>(para_id: ParaId) -> Option<Constraints<BlockNumberFor<T>>> {
```

Comment:
would be better to have some helper function that both this new API and the old one call under the hood to deduplicate the code.